data/wikipron/src/scrape.py

```diff
@@ -18,23 +18,57 @@
 from codes import LANGUAGES_PATH, LOGGING_PATH
 
 
+def _whitelist_reader(path: str) -> Iterator[str]:
+    # Read-in whitelist file.
+    with open(path, "r") as source:
+        for line in source:
+            line = re.sub(r"\s*#.*$", "", line)  # Removes comments from line.
+            yield line.rstrip()
+
+
+def _filter(word: str, pron: str, phones: FrozenSet[str]) -> bool:
+    # Determines if gloss is valid given phone set.
+    these_phones = frozenset(pron.split())
+    bad_phones = these_phones - phones
+    if bad_phones:
+        for phone in bad_phones:
+            logging.warning("Bad phone:\t%s\t(%s)", phone, word)
+        return False
+    else:
+        return True
+
+
 def _call_scrape(
-    lang_settings: Dict[str, str], config: wikipron.Config, tsv_path: str
+    lang_settings: Dict[str, str],
+    config: wikipron.Config,
+    tsv_path: str,
+    whitelist_set: FrozenSet[str] = None,
+    tsv_path_filtered: str = "",
 ) -> None:
     for unused_retries in range(10):
         with open(tsv_path, "w") as source:
             try:
-                for (word, pron) in wikipron.scrape(config):
-                    print(f"{word}\t{pron}", file=source)
+                scrape_results = wikipron.scrape(config)
+                # Given whitelist, opens up a second tsv for scraping.
+                if whitelist_set:
+                    with open(tsv_path_filtered, "w") as source_filtered:
+                        for (word, pron) in scrape_results:
+                            if _filter(word, pron, whitelist_set):
+                                print(f"{word}\t{pron}", file=source_filtered)
```

Comment:
This is pretty minor but rather than building this string to print twice, I would maybe do something like:

    line = f"{word}\t{pron}
    if _filter(...):
        print(line, file=...)
    print(line)